src/sentry/profiles/consumer.py

```diff
@@ -0,0 +1,143 @@
+        "device_os_name": profile["device_os_name"],
+        "device_os_version": profile["device_os_version"],
+        "duration_ns": int(profile["duration_ns"]),
+        "environment": profile.get("environment"),
+        "platform": profile["platform"],
+        "trace_id": profile["trace_id"],
+        "transaction_name": profile["transaction_name"],
+        "version_name": profile["version_name"],
+        "version_code": profile["version_code"],
+        "retention_days": 30,
+    }
+
+    if profile["platform"] == "android":
+        normalized_profile.update(
+            {
+                "android_api_level": profile["android_api_level"],
+                "profile": profile["stacktrace"],
+                "symbols": profile["android_trace"],
+            }
+        )
+    elif profile["platform"] == "ios":
+        normalized_profile.update(
+            {
+                "device_os_build_number": profile["device_os_build_number"],
+                "profile": profile["sampled_profile"],
+                "symbols": profile["debug_meta"],
+            }
+        )
+
+    return normalized_profile
+
+
+def _validate_ios_profile(profile: MutableMapping[str, Any]) -> bool:
+    return "sampled_profile" in profile and "samples" in profile["sampled_profile"]
+
+
+class ProfilesWorker(AbstractBatchWorker):  # type: ignore
+    def __init__(self, producer: Producer) -> None:
+        self.__producer = producer
+        self.__producer_topic = "processed-profiles"
+
+    def process_message(self, message: Any) -> MutableMapping[str, Any]:
+        message = msgpack.unpackb(message.value(), use_list=False)
+        profile = json.loads(message["payload"])
+
+        profile.update(
+            {
+                "organization_id": message["organization_id"],
+                "project_id": message["project_id"],
+                "received": message["received"],
+            }
+        )
+
+        if profile["platform"] == "ios":
+            if not _validate_ios_profile(profile):
+                return None
+            profile = self.symbolicate(profile)
+        return _normalize(profile)
+
+    def symbolicate(self, profile: MutableMapping[str, Any]) -> MutableMapping[str, Any]:
+        samples = profile["sampled_profiles"]["samples"]
+
+        # collect all unsymbolicated frames
+        frames_by_address = {}
+        indexes_by_address = defaultdict(list)
+        for i, s in enumerate(samples):
+            for j, f in enumerate(s["frames"]):
+                indexes_by_address[f["instruction_addr"]].append((i, j))
+                frames_by_address[f["instruction_addr"]] = f
+
+        # set proper keys for process_payload to do its job
+        profile["platform"] = "native"
+        profile["stacktraces"] = {"frames": frames_by_address.values()}
+        profile["event_id"] = profile["transaction_id"]
+        profile["project"] = profile["project_id"]
+
+        # symbolicate
+        profile = process_payload(profile)
```

Comment:
Maybe someone else can weigh in here but is calling `process_payload` on each profile synchronously within the consumer a good idea here?

Reply:
I mean, it's probably not going to scale very well as we get more traffic, but right now, our volume is 0 message so I feel pretty good about it.

Reply:
Do you know how you'll solve this once the scale goes up? For errors we spawn celery tasks to perform symbolication async and once symbolication is complete we resume the event pipeline. It might be good to plan ahead for when you will need to spawn tasks to symbolicate.

Reply:
I think we'll do the same thing here. I'm not quite sure if we'll need to spawn a new pool of workers for this task or if we'll just reuse the celery task you have and that's also why I punted that for later.

Reply:
Turns out I implemented the Celery queue already.